### ISSUE TITLE
Improve a11y of TOC component

### ISSUE THREAD

**Issue by DavieReid** [jpmorganchase/mosaic#315]
A few accessibility issues with the TOC component:

- every item has the same aria-level (which I think should be inline with the heading level)
- the items are `<p>` when the should be `<a>`
- not accessible via the keyboard (tabindex is -1)
